File: capstone/Backend/assesment-portal/src/main/java/com/project/assesmentportal/services/QuizService.java
package com.project.assesmentportal.services;

import java.util.List;

import com.project.assesmentportal.dto.ApiResponse;
import com.project.assesmentportal.dto.QuestionDto;
import com.project.assesmentportal.dto.QuizDto;

/**
 * Service interface for managing quiz-related operations.
 */
public interface QuizService {

    /**
     * Adds a new quiz.
     * @param quizDto The QuizDto representing the quiz to be added.
     * @return The QuizDto of the added quiz.
     */
    ApiResponse addQuiz(QuizDto quizDto);

    /**
     * Retrieves a list of all quizzes.
     * @return A list of QuizDto objects representing all quizzes.
     */
    List<QuizDto> getQuizzes();

    /**
     * Retrieves quiz by id.
     * @param quizId of existing quiz.
     * @return QuizDto of specified quizId.
     */
    QuizDto getQuizById(long quizId);

    /**
     * Updates an existing quiz.
     * @param quizDto QuizDto of updated quiz.
     * @param quizId of existing quiz.
     * @return QuizDto of updated quiz.
     */
    ApiResponse updateQuiz(QuizDto quizDto, long quizId);

    /**
     * delete quiz.
     * @param quizId of existing quiz.
     */
    ApiResponse deleteQuiz(long quizId);

    /**
     * Gets list of questions for a quiz.
     * @param quizId id of the quiz.
     * @return list of questions.
     */
    List<QuestionDto> getQuestionsByQuiz(long quizId);
}

File: capstone/Frontend/src/Services/QuizService.js
import axios from 'axios';

const QUIZ_BASE_URL = 'http://localhost:8080/quiz';

class QuizService{

    addQuiz(Quiz){
        return axios.post(QUIZ_BASE_URL, Quiz);
    }

    getAllQuizzes(){
        return axios.get(QUIZ_BASE_URL);
    }

    getQuizById(id){
        return axios.get(QUIZ_BASE_URL+'/'+id);
    }

    updateQuiz(id, Quiz){
        return axios.put(QUIZ_BASE_URL+'/'+id,Quiz);
    }

    deleteQuiz(id){
        return axios.delete(QUIZ_BASE_URL+'/'+id);
    } 

    getQuestionsByQuiz(id){
        return axios.get(QUIZ_BASE_URL+'/'+id+'/'+"questions");
    }
}

export default new QuizService();